src/main/kotlin/uk/bfi/uvaudit/event/AuditEventController.kt

```diff
@@ -39,6 +48,15 @@ class AuditEventController(
         @RequestHeader("X-API-RequestType") requestType: String,
         @RequestHeader("X-API-RequestURI") requestUri: String
     ) {
+        validateUser(user)
         writer.write(user.id, requestType, requestUri)
     }
+
+    private fun validateUser(
+        user: AuditUser
+    ) {
+        if (user.getAttribute<Boolean>("email_verified") != true) {
```

Comment:
Is there a subtlety I'm missing that means this won't work?

```
if (!user.getAttribute<Boolean>("email_verified"))
```

Reply:
`getAttribute<T>` returns a `T?` (i.e. nullable), so you'd need a non-null assertion on the result of the call first to write it like that.